src/api/operator/numpy/np_init_op.cc

```diff
@@ -216,7 +217,7 @@ MXNET_REGISTER_API("_npi.arange")
   param.repeat = 1;
   param.infer_range = false;
   if (args[3].type_code() == kNull) {
-    param.dtype = mshadow::kFloat32;
+    param.dtype = mxnet::common::GetDefaultDtype();
```

Comment:
I just realized that in numpy, the default dtype of arange is int64.
```python
import numpy as np
print(np.arange(10).dtype)
import mxnet as mx
mx.npx.set_np()
print(mx.np.arange(10).dtype)
```
Output:
```
int64
float32
```

Thus, we should change the dtype to be consistent with the official numpy. What do you think @yzhliu  @leezu

Reply:
> I just realized that in numpy, the default dtype of arange is int64.
> 
> ```python
> import numpy as np
> print(np.arange(10).dtype)
> import mxnet as mx
> mx.npx.set_np()
> print(mx.np.arange(10).dtype)
> ```
> 
> Output:
> 
> ```
> int64
> float32
> ```
> 
> Thus, we should change the dtype to be consistent with the official numpy. What do you think @yzhliu @leezu

I think maybe we could consider the most common use cases to decide which return result to choose.

Reply:
In fact, arange is usually used for constructing the index. Sometimes, the user may use `np.arange(0, 1000*1000*1000, 10000)` and we will lose precision if we use float32.

```python
import torch as th

print(th.arange(10).dtype)
# torch.int64
```

Reply:
The example @sxjscience listed looks reasonable to me. I guess we should do int64 then.